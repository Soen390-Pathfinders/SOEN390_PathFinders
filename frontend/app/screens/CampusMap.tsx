//Campus Map Screen
import { StyleSheet, Text, View, TextInput, Image } from "react-native";
import OutdoorMap from "../components/maps/OutdoorMap";
import AppLogo from "../components/ui/AppLogo";
import { ThemeProvider } from "../components/context/ThemeContext";
import ThemeToggle from "../components/ui/ThemeToggle";
import useTheme from "../hooks/useTheme";
import React, { useState } from 'react';
import { getStyles } from "../styles";
import CampusPilotHeader from "../components/ui/CampusPilotHeader";
import { CampusToggle } from "../components/ui/CampusToggle";

export default function CampusMap() {
  const { theme, toggleTheme } = useTheme(); // Use the custom hook
  const styles = getStyles(theme); // Get styles based on theme

  const [campus, setCampus] = useState("SGW"); // Default to SGW

  const toggleCampus = (selectedCampus) => {
    setCampus(selectedCampus);
  };
  return (
    <View style={styles.container}>
      {/* Header with logo , title and Dakr/Light theme toggle */}
     <View style={{ marginTop: -150 }}>
               <CampusPilotHeader />
      </View>
      <CampusToggle campus={campus} toggleCampus={toggleCampus} />
      {/*SearchBar
      <View style={styles.searchBar}>
        <TextInput placeholder={"Search"} placeholderTextColor={"#666"} />
      </View>*/}

      {/*Campus Map*/}
      <View style={styles.mapContainer}>
        <OutdoorMap campus = {campus}/>
      </View>
    </View>
  );
}
