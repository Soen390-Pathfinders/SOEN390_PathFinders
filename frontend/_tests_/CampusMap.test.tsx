import React from 'react';
import { render } from '@testing-library/react-native';
import CampusMap from '../app/screens/CampusMap';

// Mock the useTheme hook to always return light theme
jest.mock('../app/hooks/useTheme', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    theme: 'light',
    toggleTheme: jest.fn(),
  })),
}));

// Mock getStyles to return dummy styles
jest.mock('../app/styles', () => ({
  getStyles: (theme: string) => ({
    container: { flex: 1, backgroundColor: theme === 'light' ? '#fff' : '#000' },
    mapContainer: { flex: 1 },
  }),
}));

// Mock child components so we isolate CampusMap rendering
jest.mock('../app/components/ui/CampusPilotHeader', () => () => <></>);
jest.mock('../app/components/ui/CampusToggle', () => ({
  CampusToggle: ({ campus, toggleCampus }: any) => <></>,
}));
jest.mock('../app/components/maps/OutdoorMap', () => () => <></>);

describe('CampusMap Screen', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<CampusMap />);
    expect(toJSON()).toMatchSnapshot();
  });
});
