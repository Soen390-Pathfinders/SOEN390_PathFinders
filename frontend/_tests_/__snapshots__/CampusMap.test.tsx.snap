// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampusMap Screen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;
